modules/client/src/main/java/org/apache/ignite/internal/client/sql/ClientSession.java

```diff
@@ -82,15 +83,17 @@ public ClientSession(
     /** {@inheritDoc} */
     @Override
     public ResultSet execute(@Nullable Transaction transaction, String query, @Nullable Object... arguments) {
-        // TODO IGNITE-17057.
-        throw new UnsupportedOperationException("Not implemented yet.");
+        Objects.requireNonNull(query);
+
+        return new ResultSetImpl(sync(executeAsync(transaction, query, arguments)));
```

Comment:
Different methods are used to await CompletableFuture at three places: 
- sync (here)
- await (embedded SessionImpl)
- join (ResultSetImpl)

What do you think about unification?

Reply:
What do you think about default implementation at the Session interface?

Reply:
I guess `CompletionException` is not expected by user at the synchronous API.
Could we make temporary sync/await method with convert to SqlException?

Reply:
Agree, added a temporary fix.